build.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh
+
+# exit when any command fails
+set -e
+
+p() {
+  printf "\e[33m$1\e[0m\n"
+}
+
+function _build() {
+
+  # 1. Define variables for use in the script
+  team_name=pq-team
+  ecr_repo_name=parliamentary-questions
+  component=pq-tracker
+
+  region='eu-west-2'
+  context='live-1'
+  aws_profile='ecr-live-1-parliamentary-questions'
+
+  git_remote_url="https://github.com/ministryofjustice/parliamentary-questions.git";
+  docker_endpoint=754256621582.dkr.ecr.eu-west-2.amazonaws.com
+  docker_registry=${docker_endpoint}/${team_name}/${ecr_repo_name}
+
+  current_branch=$(git branch | grep \* | cut -d ' ' -f2)
+  current_version=$(git rev-parse $current_branch)
+  short_version=$(git rev-parse --short $current_branch)
+
+  docker_build_tag=pf-${current_branch}-${short_version}
+  docker_registry_tag=${docker_registry}:${docker_build_tag}
+
+  # 2. Display status message - include warning if the working copy is not clean
+  p "------------------------------------------------------------------------"
+  p "Building PQ Tracker image for deployment"
+  p "Git repository: $git_remote_url"
+  p "Build tag: $docker_build_tag"
+  p "Branch: $current_branch"
+  p "Registry tag: $docker_registry_tag"
+
+  if [ -z "$(git status --porcelain)" ]; then
+    p "Deploying from a clean working directory..."
+  else
+    p "\e[31mWarning! Working directory contains new or modified files...\e[0m\n"
+    git status --porcelain
+  fi
+  p "------------------------------------------------------------------------"
+
+  # 3. Get a logged in context so we can push images to the ECR
+  p "Docker login to registry (ECR)..."
+  $(aws ecr --profile "$aws_profile" get-login --no-include-email --region "$region" --profile "$aws_profile")
```

Comment:
Nick, this is the bit I was talking about earlier in passing - never bothered to stop and think about it, but the `$(...)` here is the equivalent of the copy/paste you were doing when getting the authentication to the ECR - it generates the command and executes it in one statement.